ship/src/main/scala/ch/epfl/bluebrain/nexus/ship/RunShip.scala

```diff
@@ -88,11 +93,28 @@ trait RunShip {
 object RunShip {
 
   def localShip = new RunShip {
+    override def loadConfig(config: Option[Path]): IO[ShipConfig] =
+      ShipConfig.load(config)
+
     override def eventsStream(path: Path, fromOffset: Offset): Stream[IO, RowEvent] =
       EventStreamer.localStreamer.stream(path, fromOffset)
   }
 
   def s3Ship(client: S3StorageClient, bucket: BucketName) = new RunShip {
+    override def loadConfig(config: Option[Path]): IO[ShipConfig] = {
+      config match {
+        case Some(value) =>
+          val stream = client.readFile(bucket, FileKey(NonEmptyString.unsafeFrom(value.toString)))
+          val path   = Path("/tmp/ship/s3ship-external.conf")
+          Files[IO].writeAll(path)(stream).compile.drain.flatMap { _ =>
+            val configFile = new File(path.toString)
+            ShipConfig.loadFromFile(Some(configFile))
+          }
```

Comment:
Using temporary file location for the config since I don't think we can use the stream directly to load the config (?)